packages/story-editor/src/components/canvas/utils/useFullbleedMediaAsBackground.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import isTargetCoveringContainer from '../../../utils/isTargetCoveringContainer';
+import { useStory, useCanvas } from '../../../app';
+
+function useFullbleedMediaAsBackground({ selectedElement }) {
+  const { setBackgroundElement, currentPage } = useStory((state) => ({
+    setBackgroundElement: state.actions.setBackgroundElement,
+    currentPage: state.state.currentPage,
+  }));
+  const { fullbleedContainer } = useCanvas(
+    ({ state: { fullbleedContainer } }) => ({
+      fullbleedContainer,
+    })
+  );
+
+  const handleFullbleedMediaAsBackground = (target) => {
+    if (!currentPage?.elements[0]?.isDefaultBackground) {
+      return false;
+    }
+    if (selectedElement.type !== 'image' && selectedElement.type !== 'video') {
```

Comment:
Could also be `gif`.

Reply:
Is it worth tieing to `const DROP_SOURCE_ALLOWED_TYPES = ['image', 'gif', 'video'];` (`DROP_SOURCE_ALLOWED_TYPES.includes(selectedElement.type)`)?

Reply:
Yep, maybe even renaming it to `MEDIA_ELEMENT_TYPES` or something similar, and moving into the general constants file (we have `constants/media.js` for example)? Looks like it's used only in one place currently so would be a small change.

Reply:
Should be fine https://github.com/GoogleForCreators/web-stories-wp/pull/10458/commits/8601f296e9df69e05a92d96b0809119a5010203a ?
I will do the local change next once you are ok with this^.

Reply:
Yep, lgtm!

Reply:
👍🏻  https://github.com/GoogleForCreators/web-stories-wp/pull/10458/commits/75112b51f4b56c16021396fbc06e57c3e74fa7c3